cli/pcluster/config/validators.py

```diff
@@ -1332,3 +1320,33 @@ def ebs_volume_iops_validator(section_key, section_label, pcluster_config):
             )
 
     return errors, warnings
+
+
+def ebs_volume_size_snapshot_validator(section_key, section_label, pcluster_config):
+    """
+    Validate the following cases.
+
+    The EBS snapshot is in "completed" state if it is specified
+    If user specified the volume size, the volume must be larger than then volume size of the EBS snapshot
+    """
+    errors = []
+    warnings = []
+
+    section = pcluster_config.get_section(section_key, section_label)
+    if section.get_param_value("ebs_snapshot_id"):
+        ebs_snapshot_id = section.get_param_value("ebs_snapshot_id")
```

Comment:
Since we're covering with this validator the check for whether or not the snapshot exists, it would be nice to catch an exception for it here and handle that case explicitly:

```python
try:
    ebs_snapshot_id = section.get_param_value("ebs_snapshot_id")
    snapshot_response_dict = get_ebs_snapshot_info(ebs_snapshot_id, raise_exceptions=True)
except Exception as exception:
    if isinstance(exception, ClientError) and exception.response.get("Error").get("Code") in ["InvalidSnapshot.NotFound", "InvalidSnapshot.Malformed"]:
        pass  # append to errors some message about the snapshot not existing
    else:
        pass  # append to errors some generic message about being unable to get info for the snapshot
```

Note that this also requires implementing the optional `raise_exceptions` arg in `get_ebs_snapshot_info`.

Reply:
Good idea. Done